test/remote/gateways/remote_sum_up_test.rb

```diff
@@ -123,6 +124,30 @@ def test_failed_void_invalid_checkout_id
     assert_equal 'Resource not found', response.message
   end
 
+  # In Sum Up the account can only return checkout/purchase in pending or success status,
+  # to obtain a successful refund we will need an account that returns the checkout/purchase in successful status
+  #
+  # For this example configure in the fixtures => :sum_up_account_for_successful_purchases
+  def test_successful_refund
+    gateway = SumUpGateway.new(fixtures(:sum_up_account_for_successful_purchases))
+    purchase = gateway.purchase(@amount, @credit_card, @options)
+    transaction_id = purchase.params['transaction_id']
```

Comment:
We should `assert` the transaction_id here, otherwise the test throws an error if the value is `nil`

Reply:
Done!